src/py42/_internal/archive_access.py

```diff
@@ -125,61 +171,151 @@ def _get_children(self, node_id=None):
+        if not timeout:
+            # Skips file size calculation
+            return None
+
+        job_ids = self._start_poll(file_ids)
+        return self._wait_for_jobs(job_ids, timeout)
+
+    def _start_poll(self, file_ids):
+        job_ids = []
+        for file_id in file_ids:
+            response = self._storage_archive_client.create_file_size_job(
+                self._device_guid, file_id
+            )
+            job_id = response[u"jobId"]
+            job_ids.append(job_id)
+        return job_ids
+
+    def _wait_for_jobs(self, job_ids, timeout):
+        t0 = time.time()
+        sizes = []
+
+        # Waits until the job_ids stack is empty
+        while job_ids:
+            for job_id in job_ids:
+                response = self._get_job_status(job_id)
+                size_dict = _create_size_dict(job_id, response)
+                _print_file_size(size_dict)
+                if response[u"status"] == u"DONE":
+                    job_ids.remove(job_id)
+                    sizes.append(size_dict)
+
+            # File size calculation is taking too long.
+            if time.time() - t0 > timeout:
+                return None
+
+            if job_ids:
+                time.sleep(self._job_polling_interval)
+        return sizes
+
+    def _get_job_status(self, job_id):
+        return self._storage_archive_client.get_file_size_job(job_id, self._device_guid)
+
+
+class RestoreJobManager(_RestorePoller):
     def __init__(
         self,
         storage_archive_client,
         device_guid,
         archive_session_id,
-        job_polling_interval=JOB_POLLING_INTERVAL_SECONDS,
+        job_polling_interval=None,
     ):
-        self._storage_archive_client = storage_archive_client
-        self._device_guid = device_guid
+        super(RestoreJobManager, self).__init__(
+            storage_archive_client, device_guid, job_polling_interval
+        )
         self._archive_session_id = archive_session_id
-        self._job_polling_interval = job_polling_interval
 
-    def get_stream(self, file_selection):
-        response = self._start_restore(file_selection)
+    def get_stream(self, file_selections):
+        response = self._start_restore(file_selections)
         job_id = response["jobId"]
+        self._wait_for_job(job_id)
+        return self._get_stream(job_id)
 
-        while not self.is_job_complete(job_id):
+    def _wait_for_job(self, job_id):
+        while not self._is_job_complete(job_id):
             time.sleep(self._job_polling_interval)
 
-        return self._get_stream(job_id)
-
-    def is_job_complete(self, job_id):
+    def _is_job_complete(self, job_id):
         response = self._storage_archive_client.get_restore_status(job_id)
-        return self._get_completion_status(response)
+        percentage_dict = {
+            u"jobId": job_id,
+            u"percentComplete": response[u"percentComplete"],
+        }
+        debug.logger.debug(format_dict(percentage_dict))
```

Comment:
logging percent complete now! @timabrmsn

Reply:
It looks like there are two parts to the web restore job that have different percentages. The status "preparing" and the status "compressing". We should probably add status to the percentage dict so it doesn't look like the job just starts over at 0% and runs through again.